components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -671,7 +671,14 @@ def canChgrp(self):
         group. Web client will only allow this for the data Owner. Admin CAN
         move other user's data, but we don't support this in Web yet.
         """
-        return self.isOwned() or self._conn.isAdmin()   # See #8974
+        return self.getDetails().getPermissions().canChgrp()
```

Comment:
cc: @will-moore so he can come back and raise any issues he realizes this may affect.